docs/pages/database-access/enroll-google-cloud-databases/spanner.mdx

```diff
@@ -0,0 +1,218 @@
+---
+title: Database Access with Cloud Spanner
+description: How to configure Teleport database access with GCP's Cloud Spanner.
+---
+
+(!docs/pages/includes/database-access/db-introduction.mdx dbType="Cloud Spanner" dbConfigure="with a service account"!)
+
+## How it works
+
+(!docs/pages/includes/database-access/how-it-works/iam.mdx db="Spanner" cloud="Google Cloud"!)
+
+<Tabs>
+
+<TabItem label="Self-Hosted">
+![Teleport Cloud Spanner Self-Hosted](../../../img/database-access/guides/spanner_selfhosted.png)
+</TabItem>
+
+<TabItem label="Teleport Enterprise (cloud-hosted)">
+![Teleport Cloud Spanner Cloud-Hosted](../../../img/database-access/guides/spanner_cloud.png)
+</TabItem>
+
+</Tabs>
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+
+- Google Cloud account
+- Google Cloud Spanner database.
+- A host where you will run the Teleport Database Service, e.g. a Compute Engine
+  instance.
+- (!docs/pages/includes/tctl.mdx!)
+
+## Step 1/8. Create a service account for the Teleport Database Service
+
+Teleport uses [IAM authentication](https://cloud.google.com/spanner/docs/iam)
+to access the Cloud Spanner API.
+
+(!docs/pages/includes/database-access/cloudsql-create-service-account-for-db-service.mdx !)
+
+### Create a key for the service account
+
+(!docs/pages/includes/database-access/cloudsql_service_account_key.mdx!)
+
+## Step 2/8. Create a service account for a database user
+
+When a user connects to Spanner via Teleport, they will specify a service
+account name as the "database user" and the Teleport Database service will
+impersonate that service account to authenticate to GCP.
+
+You can create multiple service accounts to provide different access levels to
+Teleport users, but for this guide we will just create one. 
+
+(!docs/pages/includes/database-access/cloudsql_create_db_user_account.mdx!)
+
+### Grant permissions
+
+Click "select a role", choose the `Cloud Spanner Database User` role, and
+then click "Done":
+
+![Grant Cloud Spanner Database User to Service Account](../../../img/database-access/guides/cloudsql/grant-cloud-spanner-user@2x.png)
+
+<Admonition type="note">
```

Comment:
I think this Admonition is a lot of information to take in for a user who's trying to get through this guide in minimal time. Is there a way to incorporate the principle of least privilege into this section, e.g., have the user define a role with the same policy as `Cloud Spanner Database User`, but scoped to specific databases?

Reply:
Good idea - I'll update the permission binding scope to a specific database, but I'll keep using the pre-defined role since it's a basic Spanner user pre-defined role.

Reply:
See: [8bff061](https://github.com/gravitational/teleport/pull/41804/commits/8bff061bac2c0712efc1f3363aa7fa84abd3447f)